web-client/src/views/DocketRecord/DocketRecord.tsx

```diff
@@ -9,27 +9,42 @@ import { UnsealDocketEntryModal } from './UnsealDocketEntryModal';
 import { connect } from '@web-client/presenter/shared.cerebral';
 import { sequences } from '@web-client/presenter/app.cerebral';
 import { state } from '@web-client/presenter/app.cerebral';
-import React from 'react';
+import React, { useEffect } from 'react';
 import classNames from 'classnames';
 
 export const DocketRecord = connect(
   {
+    caseDetail: state.caseDetail,
     docketRecordHelper: state.docketRecordHelper,
-    formattedDocketEntries: state.formattedDocketEntries,
+    formattedDocketEntriesHelper: state.formattedDocketEntries,
     openSealDocketEntryModalSequence:
       sequences.openSealDocketEntryModalSequence,
     openUnsealDocketEntryModalSequence:
       sequences.openUnsealDocketEntryModalSequence,
+    setSelectedDocumentsForDownloadSequence:
+      sequences.setSelectedDocumentsForDownloadSequence,
     showModal: state.modal.showModal,
   },
 
   function DocketRecord({
     docketRecordHelper,
-    formattedDocketEntries,
+    formattedDocketEntriesHelper,
     openSealDocketEntryModalSequence,
     openUnsealDocketEntryModalSequence,
+    setSelectedDocumentsForDownloadSequence,
     showModal,
   }) {
+    useEffect(() => {
+      if (!docketRecordHelper.showBatchDownloadControls) return;
+
+      const documentsSelectorHeaderInput = window.document.getElementById(
+        'all-selectable-docket-entries-checkbox',
+      ) as HTMLInputElement;
+      if (formattedDocketEntriesHelper.someDocumentsSelectedForDownload) {
+        documentsSelectorHeaderInput.indeterminate = true;
+      } else documentsSelectorHeaderInput.indeterminate = false;
+    }, [formattedDocketEntriesHelper.someDocumentsSelectedForDownload]);
```

Comment:
Not super important: since we don't have a React testing framework, ideally most of the display logic would be done in a computed. I don't know if it's possible to bind documentsSelectorHeaderInput.indeterminate to state, but that would allow this logic to be tested